### DISCUSSION TITLE
Settings > Security > Max Upload Size & Max Files per Upload options don't work

### DISCUSSION THREAD

**Discussion by ProviceUnify** [requarks/wiki#7077] [Error / Bug Report] [answered]
Parameters `Max Upload Size` and `Max Files per Upload` don't have any affect on real top upload limits.
Tried to set options in:
- `Settings` > `Security`
- By deprecated method in config.yml
- Right in database (was set before checked this)

But didn't get any changes. Files less 10pcs and under 5mb are uploading, more - no.

Connection is plain - no reverse proxy (e.g. Nginx, Apache).
DB: MariaDB (MySQL)
Wiki.js version: 2.5.301
OS: Windows Server

![image](https://github.com/requarks/wiki/assets/22263818/367b9cab-59d8-46ff-b81b-545a0243a838)
![image](https://github.com/requarks/wiki/assets/22263818/e14e38fc-5af5-4724-9665-d405963cd517)
![image](https://github.com/requarks/wiki/assets/22263818/ef62fc87-dbc5-486d-9ca9-56e1cdd0e09d)

**Comment by ProviceUnify** [comment]
Checked it again on home PC - limits affect on max file size. Have to recheck value on work PC. Will report back and close topic if i was wrong

**Comment by NGPixel** [comment]
Make sure to restart Wiki.js after changing values.

**Reply by ProviceUnify** [reply]
Not neccessary. Works right after "Apply" button hitted

**Comment by ProviceUnify** [comment, marked as answer]
Ok, my bad. Problem was in mistake during to size conversion. I used MegaBITS instead of MegaBYTES in Calculator to get bytes and got 18MB instead of 150MB as i wanted :D Just my inattention, sorry.

Also, you may be helpful to check this section https://docs.requarks.io/troubleshooting if you got simillar issue

**Reply by ProviceUnify** [reply]
Suggestion: use MB instead of Bytes for the text field to more comfy UI

**Comment by thamow** [comment]
For anyone that might stumble across this in the future: 
I did have a similar issue and tho it is not really resolved yet, i seem to be able to upload larger files now. One problem is that despite the settings being stored in the db correctly, the frontend continues to show the default limits when uploading a file. This should not effect the upload process tho so go ahead and ignore it. The issue for me was actually user permissions:

    1. Change the file upload size/number limit in the admin settings (located under "security" for whatever reason). Make sure to click on "Apply" at the top when finished!
    2. Edit any page and try to upload a file larger than the default 5 MB. Take note of the path that you are trying to upload to (not to be mistaken with the path of the file currently editing! Take a look at the path in the upload pop-up window at the top, it should say '/root' by default).
    3. If the upload fails, make sure that you have the needed permissions to upload assets to the aforementioned path
       ( admin settings -> Users/Groups -> your user/group -> Permissions: read/write/manage assets as well as
         admin settings -> Users/Groups -> your user/group -> Page Rules: read/write/manage assets for the upload path)
    4. If the error persists, try to add a folder to upload to (e.g. "assets"). You can do that in the upload pop-up window. Select it so that you are not uploading to /root anymore. Make sure to update the permissions / page rules for the user to include the new path if need be.

**Comment by SpringWhisper** [comment]
This is a solution:

The maximum limit for the number of uploadable files is specified in the code, rather than being dynamically retrieved from the database. 

https://github.com/requarks/wiki/blob/main/client/components/editor/editor-modal-media.vue#L145

The above link indicates the location. If the line number changes, search for "max-files". 

You can change it to a larger value, for example:

max-files='100'

Incidentally, the limit on the number of files on the UI is also written in the code. If you want to change, search for "10 files".
